components/renku_data_services/session/api.spec.yaml

```diff
@@ -907,6 +913,15 @@ components:
         - "failed"
         - "cancelled"
       example: "succeeded"
+    EnvVariables:
+      description: Environment variables for the session pod
+      type: object
+      additionalProperties:
+        type: string
+        default: {}
+      example:
+        MY_VAR: value 1
+        VAR2: null
```

Comment:
issue: You should add the validation here - this will give the nicest error message when it is not valid. The code generated from the apispec will produce nice, descriptive error messages that are also handled and returned to the caller. So you would have a schema for the key and a schema for the value. And you can combine the two into an Env variable.

When you do this apply the regex validation only for requests. For the responses we should just return str or str | null.

The same validation should remain in the `EnvVar` dataclass I proposed in the models.py file too. This is because if we relax this developers can easily forget and initialize and sneak in invalid values thorugh some other code path.